src/nnsDapp.ts

```diff
@@ -0,0 +1,98 @@
+import { Actor } from "@dfinity/agent";
+import { idlFactory as certifiedIdlFactory } from "../candid/nns_dapp.certified.idl";
+import { idlFactory, NNSDappService } from "../candid/nns_dapp.idl";
+import {
+  CreateSubAccountResponse,
+  SubAccountDetails,
+} from "../candid/nns_dappTypes";
+import { AccountIdentifier } from "./account_identifier";
+import { AccountNotFoundError } from "./errors";
+import { NNSDappCanisterOptions } from "./types/nnsDapp";
+import { defaultAgent } from "./utils/agent.utils";
+
+export class NNSDappCanister {
+  private constructor(
+    private readonly service: NNSDappService,
+    private readonly certifiedService: NNSDappService
+  ) {
+    this.service = service;
+    this.certifiedService = certifiedService;
+  }
+
+  public static create(options: NNSDappCanisterOptions) {
+    const agent = options.agent ?? defaultAgent();
```

Comment:
- possible null exception since `options` has no default value as in the other canister
- goodie: deconstruct variables and optional chaining